blockchain/src/blockchain/history_sync.rs

```diff
@@ -96,24 +96,39 @@ impl Blockchain {
     /// Extends the current chain with transactions from the validity sync process
     pub fn extend_validity_sync(
         this: RwLockUpgradableReadGuard<Blockchain>,
-        epoch: u32,
         history: &[HistoricTransaction],
     ) -> Option<Blake2bHash> {
```

Comment:
Looks to me as if the changes made this no longer fail for any reason, but panic instead.
```suggestion
    ) -> Blake2bHash {
```